### PR TITLE
Ap edit post

### PR DESCRIPTION
**Pull request by daprieu**
# Description

user is able to click button `edit` and will be taken to a form where they can edit post. If user decides not to edit user can click cancel and will be taken back to `my post`

Fixes #15

## Type of change

- [x] New feature (non-breaking change which adds functionality)

- [x] This change requires a documentation update

# Testing Instructions

`git fetch --all`
`git checkout ap-editPost`

go to my post select edit

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors
- [x] I have added test instructions that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/components/posts/PostForm.js [hunk 1]
```diff
@@ -44,15 +56,38 @@ export const PostForm = () => {
...
+    }   
+  }
+// debugger
+const urlId = postId.postId
+console.log('urlId: ', urlId);
+useEffect(() => {
+  getCategories()
+  .then(getPosts)
+  .then(posts => { 
+    const postbyId = posts.find(p => p.id === parseInt(urlId))
```
Comment (notdanross) [inline src/components/posts/PostForm.js:67]: This runs every time the form loads, so when adding a new post postById is undefined, breaking the code. I think a conditional should fix this.

**Review by notdanross — CHANGES_REQUESTED** [review]
The form breaks when trying to add a new post. See note at line 68.

**Review by notdanross — APPROVED** [review]
tested and approved. Full functionality on both uses of the form.

**Comment by notdanross** [comment]
@daprieu that conditional update looks great!

**Review by CheoR — APPROVED** [review]
Edit button does not show up if post does not belong to logged in user.

Logged in user can edit their post.

**Pull request merged**